src/components/Api.js

```diff
@@ -0,0 +1,83 @@
+export default class Api {
+  constructor(baseURL, headers) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error ${res.status}`);
+    }
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseURL}/cards`, {
+      headers: this._headers,
+    }).then(this._handleResponse);
+  }
+
+  // other methods for working with the API
+  getUserInfo() {
+    return fetch(`${this._baseURL}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    }).then(this._handleResponse);
+  }
+
+  setUserInfo(name, about) {
+    return fetch(`${this._baseURL}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      "Content-Type": "application/json",
+
+      body: JSON.stringify({
+        name,
+        about,
+      }),
+    }).then(this._handleResponse);
+  }
+
+  setUserAvatar(link) {
+    console.log("Avatar link:", link);
+
+    return fetch(`${this._baseURL}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      "Content-Type": "application/json",
+
+      body: JSON.stringify({ avatar: link }),
+    }).then(this._handleResponse);
+  }
+
+  uploadCard({ name, link }) {
+    return fetch(`${this._baseURL}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      "Content-Type": "application/json",
```

Comment:
* Please, correct my remark again https://github.com/c2yz4rp/se_project_aroundtheus/pull/1/files#r1830701836 . You don't need ` "Content-Type": "application/json",` because `headers: this._headers,` has it inside